### ISSUE TITLE
Incorrect result of GetDiagnostics() method call for Visual Basic syntax tree

### ISSUE THREAD

**Issue by Seteh** [dotnet/roslyn#13616]
Visual Basic implementation doesn't catch the `'Return' statement in a Sub or a Set cannot return a value.` diagnostic.

![Actual](http://i.imgur.com/uJIU5E8.png)

While Visual Studio does:
![Expected something like that](http://i.imgur.com/zIEZL7F.png)

**Version Used**: 14.0.25421.03 Update 3.

Code:
```C#
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.VisualBasic;
using System;
using System.Linq;

namespace ConsoleApplication2 {
    class Program {
        static void Main() {
            SyntaxTree tree = VisualBasicSyntaxTree.ParseText(@"
Imports System
Namespace TestNS
    Class TestClass
        Sub TestMethod(ByVal ar As Array)
            Return ar.GetHashCode
        End Sub
    End Class
End Namespace");

            SyntaxNode root = tree.GetRoot();
            Console.WriteLine(root.GetDiagnostics().Count());
            Console.ReadKey();
        }
    }
}
```

**Comment by svick** [comment]
`GetDiagnostics()` on a `SyntaxNode` or `SyntaxTree` only returns *syntax errors*. But this is a semantic error, so you need to create a `Compilation`:

```c#
var compilation = VisualBasicCompilation.Create(null)
    .WithOptions(new VisualBasicCompilationOptions(OutputKind.DynamicallyLinkedLibrary))
    .AddReferences(MetadataReference.CreateFromFile(typeof(object).Assembly.Location))
    .AddSyntaxTrees(tree);

foreach (var diagnostic in compilation.GetDiagnostics())
{
    Console.WriteLine(diagnostic);
}
```

This outputs:

```
(6,13): error BC30647: 'Return' statement in a Sub or a Set cannot return a value.
```

**Comment by Seteh** [comment]
@svick Thank you, :thumbsup:!
